extensions/ql-vscode/src/helpers.ts

```diff
@@ -100,6 +102,38 @@ export async function showBinaryChoiceDialog(message: string, modal = true): Pro
   return chosenItem?.title === yesItem.title;
 }
 
+/**
+ * Opens a modal dialog for the user to make a yes/no choice.
+ *
+ * @param message The message to show.
+ * @param modal If true (the default), show a modal dialog box, otherwise dialog is non-modal and can
+ *        be closed even if the user does not make a choice.
+ *
+ * @return
+ *  `true` if the user clicks 'Yes',
+ *  `false` if the user clicks 'No' or cancels the dialog,
+ *  `undefined` if the dialog is closed without the user making a choice.
+ */
+export async function showBinaryChoiceWithUrlDialog(message: string, url: string, modal = true): Promise<boolean | undefined> {
+  const urlItem = { title: 'More Information', isCloseAffordance: false };
+  const yesItem = { title: 'Yes', isCloseAffordance: false };
+  const noItem = { title: 'No', isCloseAffordance: true };
+  let chosenItem;
+
+  // Keep the dialog open as long as the user is clicking the 'more information' option.
+  do {
+    chosenItem = await Window.showInformationMessage(message, { modal }, urlItem, yesItem, noItem);
+    if (chosenItem === urlItem) {
+      await env.openExternal(Uri.parse(url, true));
+    }
+  } while (chosenItem === urlItem);
```

Comment:
Reading this I am worried about it creating dialogs in a loop.
This works for modal dialogs because VS Code doesn't create a fresh dialog each time right?

If we used this helper function in a non-modal way would you get repeated popups?

Reply:
Hmm...since we're not using this for non-modals, I'll just remove the option.

And to be safe, I'll add an escape hatch where if someone clicks the "More Info", say, 5 times, the modal is closed and value returned is "no".